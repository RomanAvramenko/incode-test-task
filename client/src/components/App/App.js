import React from "react";
import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Box, Paper } from "@mui/material";
import { WEBSOCKET_SEND } from "../../store/actionTypes";
import { Ticker } from "../Ticker/Ticker";
import "./App.scss";

function App() {
  const { prevTicker, ticker } = useSelector((store) => store.app);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch({ type: WEBSOCKET_SEND });
  }, []);

  const renderTickers = ticker.map((i, index) => {
    let comparison;
    if (prevTicker.length > 0) {
      comparison = prevTicker[index].price > i.price;
    }
    return (
      <Paper
        key={index}
        elevation={3}
        children={<Ticker data={i} comparison={comparison} />}
      />
    );
  });

  return (
    <div className="App">
      <Box
        sx={{
          display: "flex",
          justifyContent: "center",
          flexWrap: "wrap",
          margin: "1em",
          "& > :not(style)": {
            m: 1,
            width: 167,
            height: 70,
          },
        }}
      >
        {renderTickers}
      </Box>
    </div>
  );
}

export default App;
