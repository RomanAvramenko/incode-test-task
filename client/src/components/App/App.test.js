import React from 'react';
import { shallow } from "enzyme";
import * as redux from "react-redux";
import App from "./App";

const mockStore = {
  ticker: [],
  prevTicker: [],
};

const useSelectorSpy = jest.spyOn(redux, "useSelector");
const useDispatchSpy = jest.spyOn(redux, "useDispatch");
const mockDispatchFn = jest.fn();
useSelectorSpy.mockReturnValue(mockStore);
useDispatchSpy.mockReturnValue(mockDispatchFn);

describe("App component", () => {
  it("App element exist in the DOM", () => {
    const component = shallow(<App />);
    const wrapper = component.find(".App");
    expect(wrapper.length).toBe(1);
  });

  it("Should render App component", () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
});
