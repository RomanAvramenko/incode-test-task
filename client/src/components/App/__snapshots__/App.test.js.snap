// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should render App component 1`] = `
<div
  className="App"
>
  <ForwardRef(Box)
    sx={
      Object {
        "& > :not(style)": Object {
          "height": 46,
          "m": 1,
          "width": 167,
        },
        "display": "flex",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "margin": "1em",
      }
    }
  />
</div>
`;
